src/vs/workbench/contrib/comments/browser/commentsViewActions.ts

```diff
@@ -17,17 +17,23 @@ import { COMMENTS_VIEW_ID } from 'vs/workbench/contrib/comments/browser/comments
 import { FocusedViewContext } from 'vs/workbench/common/contextkeys';
 import { viewFilterSubmenu } from 'vs/workbench/browser/parts/views/viewFilter';
 
+export type CommentsSortOrder = 'resource' | 'updatedAt';
+
 const CONTEXT_KEY_SHOW_RESOLVED = new RawContextKey<boolean>('commentsView.showResolvedFilter', true);
 const CONTEXT_KEY_SHOW_UNRESOLVED = new RawContextKey<boolean>('commentsView.showUnResolvedFilter', true);
+const CONTEXT_KEY_SORT_BY = new RawContextKey<CommentsSortOrder>('commentsView.sortBy', 'resource');
+const CONTEXT_KEY_SORT_BY_UPDATED_AT = ContextKeyEqualsExpr.create('commentsView.sortBy', 'updatedAt');
 
 export interface CommentsFiltersChangeEvent {
 	showResolved?: boolean;
 	showUnresolved?: boolean;
+	sortBy?: CommentsSortOrder;
 }
 
 interface CommentsFiltersOptions {
 	showResolved: boolean;
 	showUnresolved: boolean;
+	sortBy: CommentsSortOrder;
```

Comment:
I don't think this necessarily fits perfectly here as it technically doesn't filter anything. Filtering and sorting does tend to be considered related functionality though so hopefully it's ok?